pythonCode/create_MP2RAGE_json.py

```diff
@@ -0,0 +1,136 @@
+data_MP2RAGE['DeviceSerialNumber'] = ''
+data_MP2RAGE['SoftwareVersions'] = ''
+data_MP2RAGE['StationName'] = ''
+data_MP2RAGE['InstitutionName'] = ''
+data_MP2RAGE['InstitutionAddress'] = ''
+data_MP2RAGE['InstitutionalDepartmentName'] = ''
+data_MP2RAGE['ReceiveCoilName'] = ''
+data_MP2RAGE['ReceiveCoilActiveElements'] = ''
+data_MP2RAGE['GradientSetType'] = ''
+data_MP2RAGE['MRTransmitCoilSequence'] = ''
+data_MP2RAGE['MatrixCoilMode'] = ''
+data_MP2RAGE['CoilCombinationMethod'] = ''
+data_MP2RAGE['NonlinearGradientCorrection'] = ''
+data_MP2RAGE['WaterFatShift'] = ''
+data_MP2RAGE['EchoTrainLength'] = ''
+data_MP2RAGE['DwellTime'] = ''
+data_MP2RAGE['MultibandAccelerationFactor'] = ''
+data_MP2RAGE['AnatomicalLandmarkCoordinates'] = ''
+data_MP2RAGE['MRAcquisitionType'] = ''
+data_MP2RAGE['ScanningSequence'] = ''
+data_MP2RAGE['SequenceVariant'] = ''
+data_MP2RAGE['ScanOptions'] = ''
+data_MP2RAGE['SequenceName'] = ''
+data_MP2RAGE['PulseSequenceType'] = 'MP2RAGE'
+data_MP2RAGE['PulseSequenceDetails'] = ''
+data_MP2RAGE['ParallelReductionFactorInPlane'] = ''
+data_MP2RAGE['ParallelAcquisitionTechnique'] = ''
+data_MP2RAGE['PartialFourier'] = ''
+data_MP2RAGE['PartialFourierDirection'] = ''
+data_MP2RAGE['EffectiveEchoSpacing'] = ''
+data_MP2RAGE['TotalReadoutTime'] = ''
+data_MP2RAGE['PhaseEncodingDirection'] = ''
+data_MP2RAGE['EchoTime1'] = ''  # sec
+data_MP2RAGE['EchoTime2'] = ''  # sec
+data_MP2RAGE['SliceThickness'] = ''  # mm
+
+
+'''
+WRITE THEM
+'''
+start_dir = ""  # insert here path to your BIDS data set
+
+# list all subjects
+subj_ls = next(os.walk(start_dir))[1]
+
+for iSubj in subj_ls:
+    print(iSubj)
+    # list all subfolders and files for that subject
+    subj_dir = os.walk(os.path.join(start_dir, iSubj))
+    # go through all the files for that subject
+    for path, subdirs, files in subj_dir:
+        for name in files:
+            # creates the json files in the folder where the the MP2RAGE file is found
+            if '_MPRAGE.nii.gz' in name:
+                print(os.path.join(path, name))
+
+                json_folder = path
+
+                # _inv-1_MPRAGE.json
+                json_name = name[:12] + '_inv-1_MPRAGE.json'
+                # create the file
+                with open(os.path.join(json_folder, json_name), 'w') as ff:
+                    json.dump(data_inv_1, ff, sort_keys=False, indent=4)
+
+                # _inv-2_MPRAGE.json
+                json_name = name[:12] + '_inv-2_MPRAGE.json'
+                with open(os.path.join(json_folder, json_name), 'w') as ff:
+                    json.dump(data_inv_2, ff, sort_keys=False, indent=4)
+
+                # _MPRAGE.json
+                json_name = name[:12] + '_MPRAGE.json'
+                with open(os.path.join(json_folder, json_name), 'w') as ff:
+                    json.dump(data_MP2RAGE, ff, sort_keys=False, indent=4)
+
+                # sub-01_T1map.json
+                # sub-01_T1w.json
+                data_T1['BasedOn'] = \
```

Comment:
This is a little hard to read -- could you define these strings first and then assign them as a value to this key ?